### DISCUSSION TITLE
Issues with the LG Nano clone, running displays

### DISCUSSION THREAD

**Discussion by Clarkemd** [dbuezas/lgt8fx#286] [Q&A] [answered]
Hello. I have made a prototype display device with an Arduino Nano (copy) running 3 LCD ST7735 based displays, originally using 1k resistors to level shift to 3.3v, then with a bidirectional level shifter.
Ran the displays off the nano's 3.3v out.

Got 3 displays running well, then tried a purple LGT8F328P-LQFP32 nano equivalent (I need the DAC output). Displays didn't work well, just occasional random bits shown. I found the 3.3V from the board inadequate (I believe it's supplied by the TTL/USB chip) so added a separate 3.3v supply. Found some issues with the reset line (wouldn't load sketches via USB), so I reverted from bidirectional interface to 1K resistor.

To isolate the problem as best I can, I've gone back to a single display and loaded the graphicstest example that comes with the Adafruit ST7735 & ST7789 library. I tried including SPI.begin(), though it's probably called by the Adafruit library..
I'm using the dbuezas LGZ board package, 2.0.6.
Works fine with the 328p nano, but not with the LGT nano - I get random flashes and garbled text. Possibly some issue with the LGT board support code? Setup to run 32MHz, but I've tried 16MHz.

Any ideas?

**Comment by LaZsolt** [comment]
Bad news.

It looks like `Adafruit_SPITFT.cpp` not using SPI library. Using LGT8FxP SPI library would be important because this MCU's SPI interface works quite differently than Atmel AVR SPI.
```
#define AVR_WRITESPI(x)                                                        \
  for (SPI0_DATA = (x); (!(SPI0_INTFLAGS & _BV(SPI_IF_bp)));)
```

**Comment by LaZsolt** [comment]
Good news.

By inserting a few lines, the deficiency can be corrected. https://github.com/adafruit/Adafruit-GFX-Library/pull/313

https://github.com/dbuezas/lgt8fx/wiki/Libraries-support

**Reply by Clarkemd** [reply, marked as answer]
Thanks for that! My versions of the cpp and header files are more recent than those modified, but it looks easy enough to carry the updates through. 
Just tried them and all is good. Again, thanks. The LG stuff is new to me.
Shame these updates haven't been fed back to Adafruit. Guess they haven't an interest in supporting the LG microcontrollers.
